src/providers/Participant/ParticipantProvider.tsx

```diff
@@ -20,13 +25,27 @@ const ParticipantProvider = ({ children }: Props) => {
   const { selectedOrganization } = useOrganization();
   const { activeLocale } = useLocalization();
 
-  const [currentParticipantProject, setCurrentParticipantProject] = useState<Project>();
+  const [currentParticipantProject, setCurrentParticipantProject] = useState<Project | null>();
   const [participantProjects, setParticipantProjects] = useState<Project[]>([]);
+  const [moduleProjects, setModuleProjects] = useState<Project[]>([]);
 
   const participant = useAppSelector(selectParticipant(currentParticipantProject?.participantId || -1));
   const activeModules = useAppSelector((state) => selectActiveModules(state, currentParticipantProject?.id || -1));
   const modules = useAppSelector(selectProjectModuleList(currentParticipantProject?.id || -1));
   const projects = useAppSelector(selectProjects);
+  const allModules = useAppSelector((state) =>
+    selectAllModuleList(
+      state,
+      (projects || []).map((project) => project.id)
+    )
+  );
+
+  useEffect(() => {
+    if (projects && projects.length > 0) {
```

Comment:
Some deeper issues with our selectors here. We need to refresh when our dispatch is completed here so that we can accurately determine which projects have modules, and show the screens accordingly. 

May need to rework our selectors a bit for this.

Reply:
I think we should use the async thunks for this, that way we can know when the request is completed.

Reply:
If you remove `, true` from [this](https://github.com/terraware/terraware-web/blob/main/src/redux/features/modules/modulesSlice.ts#L32), you can get the request ID and use that to determine if the request has been satisfied or not.